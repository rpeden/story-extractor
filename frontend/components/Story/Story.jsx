import React from "react";
import styles from "../../utils/styles";
import { extractDomain } from "../../utils/string-utils";

const openStoryLink = (event, link) => {
    event.preventDefault();
    window.open(link);
}

const Story = ({story}) => {
    return (
      <div key={story.address} style={{ marginBottom: "10px" }}>
        <div style={styles.story.points}>
          {story.points}
        </div>
        <div style={styles.story.title}>
          <a href={story.address}
            onClick={(event) => openStoryLink(event, story.address)}>
            {story.title}
          </a>
          <span style={styles.story.domain}> ({extractDomain(story.address)})</span>
        </div>
      </div>
    );
}

export default Story;