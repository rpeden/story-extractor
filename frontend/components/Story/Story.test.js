import React from "react";
import Story from "./Story.jsx";
import renderer from "react-test-renderer";
import {shallow} from "enzyme";

const testStory = {
  title: "Test Title",
  points: 50,
  address: "http://nowhere.com"
};

test("Story snapshot works", () => {
  const component = renderer.create(
    <Story story={testStory} />
  );
  const story = component.toJSON();
  expect(story).toMatchSnapshot();
});

test("Story renders as expected", () => {
  const story = shallow(
    <Story story={testStory} />
  );

  const storyText = story.text();
  expect(storyText).toMatch(`${testStory.points} points`);
  expect(storyText).toMatch(testStory.title);

  const storyLink = story.find("a");
  expect(storyLink.text()).toBe(testStory.title);
  expect(storyLink.prop("href")).toBe(testStory.address);
});
