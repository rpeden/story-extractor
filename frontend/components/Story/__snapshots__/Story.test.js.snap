// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story snapshot works 1`] = `
<div
  style={
    Object {
      "borderTop": "1px solid #ccc",
      "marginBottom": "10px",
      "paddingTop": "10px",
    }
  }
>
  <div
    style={
      Object {
        "display": "block",
      }
    }
  >
    <a
      href="http://nowhere.com"
      onClick={[Function]}
    >
      Test Title
    </a>
  </div>
  <div
    style={
      Object {
        "color": "#888",
        "display": "inline-block",
        "fontSize": "13px",
        "width": "150px",
      }
    }
  >
    50
     points
  </div>
  <span
    style={
      Object {
        "color": "#888",
        "display": "inline-block",
        "fontSize": "13px",
      }
    }
  >
     (
    nowhere.com
    )
  </span>
</div>
`;
